server/routes/index.ts

```diff
@@ -211,6 +211,12 @@ export default function routes(router: Router, services: Services): Router {
   post('/referrals/:id/service-categories', (req, res) => referralsController.updateServiceCategories(req, res))
   get('/referrals/:id/complexity-level', (req, res) => referralsController.viewComplexityLevel(req, res))
   post('/referrals/:id/complexity-level', (req, res) => referralsController.updateComplexityLevel(req, res))
+  get('/referrals/:referralId/service-category/:serviceCategoryId/complexity-level', (req, res) =>
+    referralsController.selectCohortComplexityLevel(req, res)
+  )
+  post('/referrals/:referralId/service-category/:serviceCategoryId/complexity-level', (req, res) =>
```

Comment:
To me it's a bit confusing to combine the GET and the POST into a single method on the referralsController.
I don't think that select is the correct wording for updating/creating a linking between complexity and service category

Reply:
The combining of GET and POST into a single method is something we've recently switched to doing on the project, so I've been following that pattern here - if it's a bit confusing, maybe it's something we should think about refactoring. I'd quite like to do that in a separate PR if possible though, so this one doesn't drag on.

Re: the term "select" - is there an alternative term you think might be more suitable? I chose "select" driven by the user interaction - the page being one for the to PP selecting the complexity level/desired outcomes on a referral (rather than a service category - this is just a way of grouping the desired outcomes/complexity level, rather than setting it on a service category).

I suppose we have `viewX` and `updateX` on other routes but I can't think of a great term that encompasses both, so I'd named the function after the user actions, like we have with `addNumberOfSessionsToActionPlan` in the `ServiceProviderReferralsController`.

Reply:
Actually, I don't mind splitting this into 2 functions now to save us from having to deal with more tech debt in future. I can refactor the desired outcomes stuff in its own PR

Reply:
That would be great. It's just something that stood out to me as a bit confusing when trying to understand it being new to the code.

Reply:
personally i prefer it combined into a single method in terms of ability to handle error conditions and readability of those code paths. but i'm not evangelical about it.

Reply:
in terms of naming - why not `getOrUpdateX`

Reply:
I think getOrUpdate would be better - even though it's not as elegant as a single verb it's more clear about the behaviour

Reply:
Ok in the interests of getting this over the line so I can start replacing existing behaviour, I'm going to leave this as I've just changed it (to two separate functions). As we'll be removing the older functions for setting desired outcomes and service categories, I might do some refactoring to try and tidy up as part of that work. Does that seem sensible?